### PR TITLE
Fix shelfmark sort

### PR DESCRIPTION
**Pull request by nonword**
Changes the `inner_hits` sort to sort on `items.shelfMark_sort` instead of `shelfMark_sort`, which appears to have resolved to the top-level `shelfMark_sort` (which exists in the qa es mapping but does not in the production mapping. probably a mistake when adding the mapping.)

We should also address why we use `shelfMark_sort` *asc* instead of *desc* when doing searches. `findByUri` sorts *desc*. Why is that different?

### CODE REVIEW

File: lib/resources.js [hunk 1]
```diff
@@ -65,7 +65,7 @@ const SEARCH_SCOPES = {
     fields: ['seriesStatement.folded']
   },
   callnumber: {
-    fields: ['shelfMark']
+    fields: ['shelfMark', 'items.shelfMark']
```
Comment (nonword) [inline lib/resources.js:68]: This is not strictly related to this fix. It just adds `items.shelfMark` as one of the fields to match on when doing a search with search_scope=callnumber, which no one uses right now. (But may soon.)

**Review by danamansana — APPROVED** [review]
We have to figure out the `asc` vs `desc` issue, but this looks good to me.

**Review by danamansana — APPROVED** [review]
